benches/bench_table.rs

```diff
@@ -36,7 +38,30 @@ fn bench_table_builder(c: &mut Criterion) {
     });
 }
 
-fn get_table_for_benchmark(count: usize) -> Table {
+/// TableGuard saves Table and TempDir, so as to ensure
+/// temporary directory is removed after table is closed.
+/// According to Rust RFC, the drop order is first `table` then
+/// `tmp_dir`.
+pub struct TableGuard {
```

Comment:
why implement it in two places?

Reply:
This struct is only used in test / bench. I think I could find a better way to use them in only test / bench while only implementing it once.

Reply:
Seems there is no way to export TableGuard only on bench. FYI, https://users.rust-lang.org/t/conditional-compilation-for-benchmarks/47227/2

> External `tests/` and `benches/` always get a non-test version of the library.

Therefore, the only way seems to be creating a `TableGuard` that will be embedded into release binary. What's your opinion?

Reply:
alright